### PR TITLE
XML Signature Validation

### PR DESCRIPTION
**Pull request by jan-cerny**
Adds ability to validate signatures in digitally signed SCAP source data streams. Validation of other XML formats than SCAP source data streams is out of scope of this patch.

Needs dependency on xmlsec library (xmlsec1-openssl-devel).

The signature validation will be performed automatically when loading the source data stream file. Users can skip it using the `--skip-signature-validation` option.

This doesn't support keys and certificates form external files, because it isn't a requirement in certification requirements. However, the implementation leaves some room for adding these options in future, because they can make sense for users.

The old tests are replaced by new more focused tests. The documentation is updated.  As OpenSCAP implements the validation of digital signatures in SCAP source data stream, using xmlsec1 is no longer described in user manual.

Fixes: #140

### CODE REVIEW

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Review by evgenyz — APPROVED** [review]
Awesome PR. That is a good start, let's merge and use it as the base for the research of corner cases. It would also be great to have it earlier for other platforms to catch build bugs (@tom-seewald I would like to kindly ask you to take a look).

File: docs/manual/manual.adoc [hunk 1]
```diff
@@ -901,62 +901,17 @@ NOTE: Please note that for the rest of `oscap` functionality, unless you specify
...
-respective elements. This is due to the fact that there are way too many options
-when it comes to keystores and crypto choices. Instead we recommend users to use
-{xmlsec}[xmlsec1] to verify their SCAP content.  Safely evaluating signed
-content (with signature verification) involves the following steps:
+When evaluating a digitally signed SCAP source data stream OpenSCAP validates
+the digital signature of the data stream. The signature validation is performed
+automatically while loading the file. Data streams with invalid signature are
+rejected and OpenSCAP doesn't evaluate the content. OpenSCAP uses
+https://www.aleksey.com/xmlsec/[XML Security Library] with OpenSSL backend to
+validate the digital signature.
```
Comment (matejak) [inline docs/manual/manual.adoc:911]:
```suggestion
validate the digital signature.

The signature validation only checks that the datastream hasn't been altered since its latest signature. OpenSCAP doesn't address trustworthiness of certificates or public keys that are part of the `KeyInfo` signature element and that are used to verify the signature. You should verify those keys yourself to prevent evaluation of datastreams that have been modified and signed by bad actors.
```

**Comment by jan-cerny** [comment]
Hi @evgenyz and @matejak , thank you for the review. I have updated the PR. What is your opinion on the test? Tell me some suggestions for testing.

**Comment by evgenyz** [comment]
I think that we are good with existing test for now. I might add another test with signature in ARF file (if it won't be a part of ARF-specific test unit, I'm still considering). But otherwise I think we should move forward with case-by-case approach.

File: utils/oscap.8 [hunk 2]
```diff
@@ -184,6 +184,11 @@ Selects a component ref from any datastream that references a component with XCC
 Do not validate input/output files.
 .RE
 .TP
+\fB\-\-skip-signature-validation\fR
```
Comment (jan-cerny) [inline utils/oscap.8:187]: @jan-cerny Please add this to `dist/bash_completion.d/oscap`.

**Comment by evgenyz** [comment]
LGTM. Thank you!
